packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/schedule.ts

```diff
@@ -0,0 +1,121 @@
+import { Duration } from '../../core';
+
+/**
+ * Schedule for EventBridge Scheduler
+ */
+export class Schedule {
+
+  /**
+   * Construct a schedule from a Date.
+   */
+  public static oneTime(time: Date): Schedule {
+    const pad = (num: number) => (num < 10 ? '0' + num : num);
+
+    const year = time.getFullYear();
+    const month = pad(time.getMonth() + 1);
+    const day = pad(time.getDate());
+    const hours = pad(time.getHours());
+    const minutes = pad(time.getMinutes());
+    const seconds = pad(time.getSeconds());
+
+    return new Schedule(`${year}-${month}-${day}T${hours}:${minutes}:${seconds}`);
+  }
+
+  /**
+   * Construct a schedule from an interval.
+   *
+   * The minimum interval is 1 minute.
+   */
+  public static rate(duration: Duration): Schedule {
+    if (duration.toMilliseconds() < Duration.minutes(1).toMilliseconds()) {
+      throw new Error('Duration cannot be less than 1 minute');
+    }
+
+    const maybeRate = (value: number, unit: string) => (value > 0 && Number.isInteger(value)) ? `${value} ${unit}` : undefined;
+
+    let rate = maybeRate(duration.toDays({ integral: false }), 'days');
+    if (rate === undefined) { rate = maybeRate(duration.toHours({ integral: false }), 'hours'); }
+    if (rate === undefined) { rate = maybeRate(duration.toMinutes({ integral: true }), 'minutes'); }
+
+    return new Schedule(`rate(${rate})`);
+  }
+
+  /**
+   * Create a schedule from a set of cron fields
+   */
+  public static cron(options: CronOptions): Schedule {
+    if (options.weekDay !== undefined && options.day !== undefined) {
+      throw new Error('Cannot supply both \'day\' and \'weekDay\', use at most one');
+    }
+
+    const minute = options.minute ?? '*';
+    const hour = options.hour ?? '*';
+    const month = options.month ?? '*';
+
+    // Weekday defaults to '?' if not supplied. If it is supplied, day must become '?'
+    const day = options.day ?? (options.weekDay !== undefined ? '?' : '*');
+    const weekDay = options.weekDay ?? '?';
+
+    const year = options.year ?? '*';
+
+    return new Schedule(`cron(${minute} ${hour} ${day} ${month} ${weekDay} ${year})`);
```

Comment:
By default, if `minute` is not provided will this job run every single minute? If so could we attach some sort of warning here?

Reply:
You're right!!

Is there a good way to display a warning? I was thinking of using Annotations.of(IConstruct).addWarning(), but since this class is not a Construct, I couldn't use it.

Reply:
Would you be able to do something similar to what is being done for the `aws-events` scheduler? https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-events/lib/schedule.ts#L64-L66

Reply:
Thanks for addressing the other comments and feedback @badmintoncryer! Just pending this warning to be added and then I'm good to approve!